crates/http/src/backend/hyper/insecure.rs

```diff
@@ -0,0 +1,88 @@
+use std::fmt::{Debug, Display};
+use std::net::SocketAddr;
+
+use scuffle_context::ContextFutExt;
+
+use crate::error::Error;
+use crate::service::{HttpService, HttpServiceFactory};
+
+#[derive(Debug, Clone)]
+pub struct InsecureBackend {
+    pub ctx: scuffle_context::Context,
+    pub bind: SocketAddr,
+    #[cfg(feature = "http1")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "http1")))]
+    pub http1_enabled: bool,
+    #[cfg(feature = "http2")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "http2")))]
+    pub http2_enabled: bool,
+}
+
+impl InsecureBackend {
+    pub async fn run<F>(self, service_factory: F) -> Result<(), Error<F>>
+    where
+        F: HttpServiceFactory + Clone + Send + 'static,
+        F::Error: Debug + Display,
+        F::Service: Clone + Send + 'static,
+        <F::Service as HttpService>::Error: std::error::Error + Debug + Display + Send + Sync,
+        <F::Service as HttpService>::ResBody: Send,
+        <<F::Service as HttpService>::ResBody as http_body::Body>::Data: Send,
+        <<F::Service as HttpService>::ResBody as http_body::Body>::Error: std::error::Error + Send + Sync,
+    {
+        #[cfg(feature = "tracing")]
+        tracing::debug!("starting server");
+
+        let listener = tokio::net::TcpListener::bind(self.bind).await?;
+
+        loop {
+            let (tcp_stream, addr) = match listener.accept().await {
```

Comment:
Before we had the ability to run multiple connection accepts from different threads, can we add that too? worker count?